xgboost.py

```diff
@@ -0,0 +1,59 @@
+import sys
+sys.path.append("..")
+
+import pandas as pd
+import numpy as np
+from xgboost import XGBClassifier
+from sklearn.model_selection import StratifiedKFold,cross_val_score
+from sklearn.metrics import f1_score,classification_report,confusion_matrix
+
+from data_preprocessing.preprocessor import Preprocessor
+
+class FraudDetector:
+    def __init__(self,df):
+        self.df = df.copy()
+        self.X_nan = None
+        self.y_nan = None
+        self.df_dropped = None
+        self.X,self.y = None,None
+        self.X_train,self.X_test,self.y_train,self.y_test = None,None,None,None
+        self.xgb = None
+        self.skf = StratifiedKFold(n_splits=5,random_state=42,shuffle=True)
+        
+    def preprocess(self):
+        preprocessor = Preprocessor(self.df)
+        preprocessor.clean()
+        
+    def extract_nan_features(self):
```

Comment:
<img src="https://avatars.githubusercontent.com/u/116506391?s=400&u=09ae00df628c1f06039a246d04cccb2b4042ed0b&v=4" alt="Image description" width="20" height="20">   CodeSweep

The `preprocess` method is currently not returning or modifying any data. It creates a `Preprocessor` object and calls the `clean` method, but the cleaned data is not stored or returned. This might lead to issues as the original dataframe remains unprocessed. Consider modifying the dataframe in-place or returning the processed dataframe.

```diff
-    def preprocess(self):
-        preprocessor = Preprocessor(self.df)
-        preprocessor.clean()
+    def preprocess(self):
+        preprocessor = Preprocessor(self.df)
+        self.df = preprocessor.clean()
```


<!-- This is an auto-generated comment by OSS CodeSweep -->